tiledb/sm/misc/utils.cc

```diff
@@ -449,6 +449,29 @@ uint64_t common_prefix_size(const std::string& a, const std::string& b) {
   return size;
 }
 
+std::string rest_components_from_url(const std::string& url) {
+  std::string path_delimiter = "/";
+  std::vector<std::string> path_parts;
+  size_t start;
+  size_t end = 0;
+  while ((start = url.find_first_not_of(path_delimiter, end)) !=
```

Comment:
This won't work because the `array_uri` component can have `/` in it. They must be escaped, see where we call `url_escape` on the array uri. This really only happens on creating of a new array where the user doesn't have a `name` but instead provides a full URI to indicate where you store the array.

See the last section of the docs for details on creation: https://docs.tiledb.com/cloud/client-api/array-access

Reply:
@ktsitsi I don't think we need to regex here. You may reconstruct the array name on L467 by looping from `i+2` to `path_parts.size()`.

Reply:
This would not work as the `url` that reaches L467 includes also operation related path parts like `deregister` , `metadata` etc. With this revision the `array_ns` and `array_uri` I think it is secured that they are the same as the ones that we get from `get_rest_components`.